plugins/chat/lib/chat/messages_exporter.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module Chat
+  module MessagesExporter
+    def chat_message_export
+      Chat::Message
+        .unscoped
+        .where(created_at: 6.months.ago..Time.current)
+        .joins(:chat_channel)
+        .joins(:user)
+        .joins("INNER JOIN users last_editors ON chat_messages.last_editor_id = last_editors.id")
+        .order(:created_at)
+        .pluck(
+          "chat_messages.id",
+          "chat_channels.id",
+          "chat_channels.name",
+          "users.id",
+          "users.username",
+          "chat_messages.message",
+          "chat_messages.cooked",
+          "chat_messages.created_at",
+          "chat_messages.updated_at",
+          "chat_messages.deleted_at",
+          "chat_messages.in_reply_to_id",
+          "last_editors.id",
+          "last_editors.username",
+        )
+    end
+
+    def get_header(entity)
+      if entity === "chat_message"
+        %w[
+          id
+          chat_channel_id
+          chat_channel_name
+          user_id
+          username
+          message
+          cooked
+          created_at
+          updated_at
+          deleted_at
+          in_reply_to_id
+          last_editor_id
+          last_editor_username
+        ]
```

Comment:
I wonder if we couldn't find a way to refactor this in a way where we a mapping defined at the top level, and on one function we use the keys and the other, the values. This way we ensure we are in sync and we don't forget to change one place when we change the other.

Reply:
Works for me :+1: